dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/QueryModifiers.java

```diff
@@ -75,4 +75,53 @@
      */
     @JsonProperty
     private final Date maxDate;
+
+    /**
+     * The sub-expression used to query the analytics value column.
+     */
+    @JsonProperty
+    private final String subExpression;
+
+    /**
+     * The value type of the sub-expression (can be different from the value
+     * type of the object).
+     */
+    @JsonProperty
+    private final ValueType valueType;
+
+    // -------------------------------------------------------------------------
+    // Logic
+    // -------------------------------------------------------------------------
+
+    /**
+     * Gets the query modifiers that matter for analytics grouping.
+     */
+    public QueryModifiers getQueryModsForAnalyticsGrouping()
```

Comment:
such modifying methods should better be named `with...` instead of `get...`

Reply:
Done.